src/main/java/com/puppycrawl/tools/checkstyle/JavaParser.java

```diff
@@ -81,14 +81,29 @@ private JavaParser() {
      */
     public static DetailAST parse(FileContents contents)
             throws CheckstyleException {
+        return parse(contents, JavaLanguageParser.CLEAR_DFA_LIMIT);
+    }
+
+    /**
+     * Static helper method to parses a Java source file.
+     *
+     * @param contents contains the contents of the file
+     * @param clearDfaLimit the number of files to parse before clearing ANTLR
+     *                      DFA states; if it is set to zero, DFA states are never
+     *                      cleared.
+     * @return the root of the AST
+     * @throws CheckstyleException if the contents is not a valid Java source
+     */
+    public static DetailAST parse(FileContents contents, int clearDfaLimit)
```

Comment:
We do not need this method, revert this change

All we need is single line update
JavaLanguageParser(tokenStream, JavaLanguageParser.CLEAR_DFA_LIMIT);

Reply:
Done. I had kept this in order to simplify test implementation, but this is not a good reason to keep this method around.